nixos/modules/services/home-automation/homebridge.nix

```diff
@@ -0,0 +1,549 @@
+    };
+
+    group = mkOption {
+      type = str;
+      default = "homebridge";
+      description = "Group to run homebridge as.";
+    };
+
+    openFirewall = mkOption {
+      type = bool;
+      default = false;
+      description = ''
+        Open ports in the firewall for the Homebridge web interface and service.
+      '';
+    };
+
+    allowInsecure = mkOption {
+      type = bool;
+      default = false;
+      description = ''
+        Allow unauthenticated requests (for easier hacking).
+        In homebridge's own installer, this is enabled by default.
+        Needs to be enabled if you want to control accessories:
+        https://github.com/homebridge/homebridge-config-ui-x/wiki/Enabling-Accessory-Control
+      '';
+    };
+
+    userStoragePath = mkOption {
+      type = str;
+      default = "/var/lib/homebridge";
+      description = ''
+        Path to store homebridge user files (needs to be writeable).
+      '';
+    };
+
+    pluginPath = mkOption {
+      type = str;
+      default = "/var/lib/homebridge/node_modules";
+      description = ''
+        Path to the plugin download directory (needs to be writeable).
+        Seems this needs to end with node_modules, as Homebridge will run npm
+        on the parent directory.
+      '';
+    };
+
+    environmentFile = mkOption {
+      type = types.nullOr types.str;
+      default = null;
+      description = ''
+        Path to an environment-file which may contain secrets.
+      '';
+    };
+
+    settings = mkOption {
+      default = { };
+      description = ''
+        Configuration options for homebridge.
+
+        For more details, see [the homebridge documentation](https://github.com/homebridge/homebridge/wiki/Homebridge-Config-JSON-Explained).
+      '';
+      type = submodule {
+        freeformType = settingsFormat.type;
+        options = {
+          description = mkOption {
+            type = str;
+            default = "Homebridge";
+            description = "Description of the homebridge instance.";
+            readOnly = true;
+          };
+
+          bridge = mkOption {
+            description = "Homebridge Bridge options";
+            default = { };
+            type = submodule {
+              freeformType = settingsFormat.type;
+              options = {
+                name = mkOption {
+                  type = str;
+                  default = "Homebridge";
+                  description = "Name of the homebridge";
+                };
+
+                port = mkOption {
+                  type = port;
+                  default = 51826;
+                  description = "The port homebridge listens on";
+                };
+              };
+            };
+          };
+
+          platforms = mkOption {
+            description = "Homebridge Platforms";
+            default = [ ];
+            apply = validatePlatforms;
+            type = listOf (submodule {
+              freeformType = settingsFormat.type;
```

Comment:
(Really not sure): you might not need to define the freeform type here

Reply:
I believe I need to set this.  I don't think `freeFormType` is inherited from the parent.  I did find another service module that specifies it on nested submodules: https://github.com/NixOS/nixpkgs/blob/1713b020bf44fbfd459638d9b13f397738da2db3/nixos/modules/services/security/canaille.nix

Reply:
Alright, really wasn't sure, best to leave it.